### PR TITLE
TB-218 OpenHIM Portal page shell

### PR DESCRIPTION
**Pull request by medchedli**
These changes require updating the DB Schema. Check  →   jembi/openhim-core-js#1212

### CODE REVIEW

File: packages/portal-app/package.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  "dependencies": {
+    "@emotion/react": "^11.10.5",
+    "@emotion/styled": "^11.10.5",
+    "@mui/icons-material": "^5.11.9",
+    "@mui/material": "^5.11.9",
+    "@types/jest": "^27.0.1",
+    "@types/react": "^17.0.19",
+    "@types/react-dom": "^17.0.9",
+    "@types/systemjs": "^6.1.1",
+    "@types/webpack-env": "^1.16.2",
```
Comment (nour-borgi) [inline packages/portal-app/package.json:60]: These should be dev dependency

File: packages/header-app/src/root.component.tsx [hunk 2]
```diff
@@ -4,18 +4,19 @@ import Box from '@mui/material/Box'
...
+import {
+  AccountCircle,
+  Menu as MenuIcon
+} from '@mui/icons-material'
+import {Divider, Drawer, IconButton, Link, Menu, MenuItem} from '@mui/material'
 import React from 'react'
 import {OpenHIMMenu} from '@jembi/openhim-sidebar'
+import LogoutIcon from '@mui/icons-material/Logout'
+import EditIcon from '@mui/icons-material/Edit'
+import {ListItemIcon} from '@mui/material'
```
Comment (MatthewErispe) [inline packages/header-app/src/root.component.tsx:16]: @medchedli can this be added to line 11?

File: packages/portal-app/src/components/AppCard/AppCard.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                  <IconButton aria-label="launch">
+                    <Button
+                      href={applicationUrl}
+                      target="_blank"
+                      rel="noopener noreferrer"
+                    >
+                      <LaunchIcon />
+                    </Button>
+                  </IconButton>
+                  /*
```
Comment (MatthewErispe) [inline packages/portal-app/src/components/AppCard/AppCard.tsx]: @medchedli do you need this?

File: packages/portal-app/src/utils/apiClient.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import axios from 'axios'
+
+const API_URL = 'https://localhost:8080/'
```
Comment (MatthewErispe) [inline packages/portal-app/src/utils/apiClient.ts:3]: @medchedli can this be configured as a evironment variable?

**Review by MatthewErispe — CHANGES_REQUESTED** [review]

File: packages/header-app/src/root.component.tsx [hunk 5]
```diff
@@ -110,13 +111,30 @@ export default function ButtonAppBar() {
...
+                  </ListItemIcon>
                   Profile
                 </MenuItem>
+                <MenuItem
+                  onClick={handleClose}
+                  component={Link}
+                  href="#!/portal"
+                >
+                  <ListItemIcon>
+                    <AppsIcon></AppsIcon>
```
Comment (bradsawadye) [inline packages/header-app/src/root.component.tsx:125]:
```suggestion
                    <AppsIcon/>
```

File: packages/header-app/src/root.component.tsx [hunk 6]
```diff
@@ -110,13 +111,30 @@ export default function ButtonAppBar() {
...
+                  Portal
+                </MenuItem>
+                <Divider />
                 <MenuItem
                   onClick={handleClose}
                   component={Link}
                   href="#!/logout"
                 >
+                  <ListItemIcon>
+                    <LogoutIcon></LogoutIcon>
```
Comment (bradsawadye) [inline packages/header-app/src/root.component.tsx:136]:
```suggestion
                    <LogoutIcon/>
```

File: packages/portal-app/.prettierignore [hunk 7]
```diff
@@ -0,0 +1,8 @@
+.gitignore
+.prettierignore
+yarn.lock
+yarn-error.log
+package-lock.json
+dist
+coverage
+pnpm-lock.yaml
```
Comment (bradsawadye) [inline packages/portal-app/.prettierignore:8]:
```suggestion
pnpm-lock.yaml

```

**Review by MatthewErispe — APPROVED** [review]

**Pull request merged**